src/hooks/connectWeb3/useWeb3React.tsx

```diff
@@ -104,12 +103,28 @@ export function ActiveWeb3Provider({
     [jsonRpcUrlMap, defaultChainId]
   )
 
+  const integratorConnection = useMemo(() => getConnectionFromProvider(onError, provider), [provider])
+
   connections = [metaMaskConnection, walletConnectConnectionQR, walletConnectConnectionPopup, networkConnection]
```

Comment:
Instead of a `useRef`, can this be memo-ized? If we memoize this, then we can also have a counter for the key with `useEffect(() => key.current += 1, [connections])`.

Reply:
I feel like everything before this line can be reverted, and the key just regenerated in this way

    const connections = useMemo(() =>
      [metaMaskConnection, walletConnectConnectionQR, walletConnectConnectionPopup, networkConnection], 
      [metaMaskConnection, walletConnectConnectionQR, walletConnectConnectionPopup, networkConnection])
    
    // Re-render the Web3ReactProvider whenever connections change, so that the passed connections is always referentially stable. This is forced using a key.
    const key = useRef(0)
    useEffect(() => key.current += 1, [connections])
    
    return (
      <Web3ReactProvider connectors={connections} key={key.current}>
        {children}
      </Web3ReactProvider>
    )